packages/editor/src/store/actions.js

```diff
@@ -602,33 +820,110 @@ export function __experimentalUpdateReusableBlockTitle( id, title ) {
 }
 
 /**
- * Returns an action object used to convert a reusable block into a static
+ * Returns an action generator used to convert a reusable block into a static
  * block.
  *
  * @param {string} clientId The client ID of the block to attach.
- *
- * @return {Object} Action object.
  */
-export function __experimentalConvertBlockToStatic( clientId ) {
-	return {
-		type: 'CONVERT_BLOCK_TO_STATIC',
-		clientId,
-	};
+export function* __experimentalConvertBlockToStatic( clientId ) {
+	const oldBlock = yield select(
+		'core/block-editor',
+		'getBlock',
+		clientId
+	);
+	const reusableBlock = yield select(
+		STORE_KEY,
+		'__experimentalGetReusableBlock',
+		oldBlock.attributes.ref
+	);
+	const referencedBlock = yield select(
+		'core/block-editor',
+		'getBlock',
+		reusableBlock.clientId
+	);
+	let newBlocks;
+	if ( referencedBlock.name === 'core/template' ) {
+		newBlocks = referencedBlock.innerBlocks.map(
+			( innerBlock ) => cloneBlock( innerBlock )
+		);
+	} else {
+		newBlocks = [ cloneBlock( referencedBlock ) ];
+	}
+	yield dispatch(
+		'core/block-editor',
+		'replaceBlocks',
+		oldBlock.clientId,
+		newBlocks
+	);
 }
 
 /**
- * Returns an action object used to convert a static block into a reusable
+ * Returns an action generator used to convert a static block into a reusable
  * block.
  *
  * @param {string} clientIds The client IDs of the block to detach.
- *
- * @return {Object} Action object.
  */
-export function __experimentalConvertBlockToReusable( clientIds ) {
-	return {
-		type: 'CONVERT_BLOCK_TO_REUSABLE',
-		clientIds: castArray( clientIds ),
+export function* __experimentalConvertBlockToReusable( clientIds ) {
+	let parsedBlock;
+	clientIds = castArray( clientIds );
+	if ( clientIds.length === 1 ) {
+		parsedBlock = yield select(
+			'core/block-editor',
+			'getBlock',
+			clientIds[ 0 ]
+		);
+	} else {
+		const blocks = yield select(
+			'core/block-editor',
+			'getBlocksByClientId',
+			clientIds
+		);
+		parsedBlock = createBlock(
+			'core/template',
+			{},
+			blocks
+		);
+
+		// This shouldn't be necessary but at the moment
+		// we expect the content of the shared blocks to live in the blocks state.
+		yield dispatch(
+			'core/block-editor',
+			'receiveBlocks',
+			[ parsedBlock ]
+		);
+	}
+
+	const reusableBlock = {
+		id: uniqueId( 'reusable' ),
+		clientId: parsedBlock.clientId,
+		title: __( 'Untitled Reusable Block' ),
 	};
+
+	yield dispatch(
+		STORE_KEY,
+		'__experimentalReceiveReusableBlocks',
+		[ { reusableBlock, parsedBlock } ]
+	);
+
+	yield dispatch(
+		'core/block-editor',
+		'replaceBlocks',
+		clientIds,
+		createBlock( 'core/block', { ref: reusableBlock.id } )
+	);
+
+	// Re-add the original block to the store, since replaceBlock() will have removed it
+	yield dispatch(
+		'core/block-editor',
+		'receiveBlocks',
+		[ parsedBlock ]
+	);
```

Comment:
**Note**, the above two dispatches (`replaceBlocks` and `receiveBlocks`) used to be _after the `_experimentalSaveReusableBlock` dispatch in the effects.  However in testing after the conversion the reusable block conversion failed because the block wouldn't actually get replaced.  I suspect this worked with effects because the timing worked a bit differently there.  Switching these around works okay.